dashboard/app/controllers/foorm/library_questions_controller.rb

```diff
@@ -5,20 +5,85 @@ class LibraryQuestionsController < ApplicationController
     before_action :authenticate_user!
     load_and_authorize_resource
 
+    # POST /foorm/library_questions
+    def create
+      library = Foorm::Library.find_by(id: params[:library_id])
+      # This needs to be put in a transaction, or another solution.
+      # If library question save fails, we end up with a created
+      # library with no associated library question.
+      if library.nil?
+        library = Foorm::Library.create(
+          {
+            name: params[:library_name],
+            version: 0,
+            published: true
+          }
+        )
+      end
+
+      @library_question.assign_attributes(
+        {
+          library_name: library.name,
+          library_version: library.version,
+          question: JSON.pretty_generate(get_question),
+          question_name: params[:name],
+          published: true
+        }
+      )
+
+      if @library_question.save
+        return render json: {
+          library_question: @library_question,
+          library: @library_question.library
+        }
+      else
+        return render status: :bad_request, json: @library_question.errors
+      end
+    end
+
     # GET /foorm/library_questions/:id
     def show
-      render json: @library_question.to_json
+      if @library_question
+        data_to_return = {
+          id: @library_question.id,
+          name: @library_question.question_name,
+          question: JSON.parse(@library_question.question),
+          published: @library_question.published
+        }
+        render json: data_to_return
+      else
+        render json: {}
+      end
     end
 
     # PUT /foorm/library_questions/:id/update
     def update
-      @library_question.question = JSON.pretty_generate(params[:question].as_json)
+      @library_question.question = JSON.pretty_generate(get_question)
```

Comment:
What happens here if `get_question` returns `render(status: :bad_request, plain: "no question provided")`, would that ever happen?

Reply:
That is a good question, I do not know...let me check.

Reply:
Yup, your suspicion is correct -- the save errors out when returning that rendered request, instead of returning a nice error message. I moved that check into a before action, so that it is returned if the question parameter isn't provided (did the same for the existing form editor).